content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -567,6 +567,8 @@ protected String buildSelectAttachmentContext(VelocityPortlet portlet, Context c
 
 		ToolSession toolSession = sessionManager.getCurrentToolSession();
 
+		context.put("maxUploadFileSize", toolSession.getAttribute(STATE_FILE_UPLOAD_MAX_SIZE));
```

Comment:
Maybe we can use `Optional<Object> maxUploadFileSize = Optional.ofNullable(toolSession.getAttribute(STATE_FILE_UPLOAD_MAX_SIZE));
` to makes sure that even if toolSession.getAttribute returns null, it's handled gracefully, reducing the risk of a NullPointerException.

Reply:
I think maybe this would not be necessary? 
In FilePickerAction.java  initHelperAction there is a null check for that attribute that populates it with a default value if it was null - see around line 1138:
`		if (toolSession.getAttribute(STATE_FILE_UPLOAD_MAX_SIZE) == null)
		{
			toolSession.setAttribute(STATE_FILE_UPLOAD_MAX_SIZE, ServerConfigurationService.getString(ResourcesConstants.SAK_PROP_MAX_UPLOAD_FILE_SIZE, 
																									  ResourcesConstants.DEFAULT_MAX_FILE_SIZE_STRING));
		}`

Reply:
Makes sense, Thanks